node-api/handlers/proof/merkle/proposer_pubkey_test.go

```diff
@@ -69,30 +70,37 @@ func TestBlockProposerPubkeyProof(t *testing.T) {
 	}
 
 	for _, tc := range testCases {
-		t.Run(tc.name, func(t *testing.T) {
-			vals := make(types.Validators, tc.numValidators)
-			for i := range vals {
-				vals[i] = &types.Validator{}
-			}
-			vals[tc.proposerIndex] = &types.Validator{Pubkey: tc.pubKey}
+		for _, forkVersion := range version.GetSupportedVersions() {
+			t.Run(tc.name+forkVersion.String(), func(t *testing.T) {
+				vals := make(types.Validators, tc.numValidators)
+				for i := range vals {
+					vals[i] = &types.Validator{}
+				}
+				vals[tc.proposerIndex] = &types.Validator{Pubkey: tc.pubKey}
 
-			bs, err := mock.NewBeaconState(
-				tc.slot, vals, 0, common.ExecutionAddress{},
-			)
-			require.NoError(t, err)
+				bs, err := mock.NewBeaconState(
+					tc.slot, vals, 0, common.ExecutionAddress{}, forkVersion,
```

Comment:
this should just take in `version.Deneb()` in this PR. https://github.com/berachain/beacon-kit/pull/2661 will add diff cases for `version.Electra()` BeaconState